contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+    uint256 public protocolFeeFactor;
+
+    /// @notice Block number that fee was last accrued at
+    uint256 public accrualBlockNumber;
+
+    /// @notice Accumulator of the total earned interest rate since start of pool
+    uint256 public utilizeIndex;
+
+    /// @notice Total amount of outstanding SD utilized
+    uint256 public totalUtilizedSD;
+
+    /// @notice Total amount of protocol fee
+    uint256 public accumulatedProtocolFee;
+
+    /// @notice utilization rate per block
+    uint256 public utilizationRatePerBlock;
+
+    /// @notice value of cToken supply
+    uint256 public cTokenTotalSupply;
+
+    /// @notice upper cap on ETH worth of SD utilized per validator
+    uint256 public maxETHWorthOfSDPerValidator;
+
+    /// @notice request ID to be finalized next
+    uint256 public nextRequestIdToFinalize;
+
+    /// @notice request ID to be assigned to a next withdraw request
+    uint256 public nextRequestId;
+
+    /// @notice amount of SD requested for withdraw
+    uint256 public sdRequestedForWithdraw;
+
+    /// @notice batch limit on withdraw requests to be finalized in single txn
+    uint256 public finalizationBatchLimit;
+
+    /// @notice amount of SD reserved for claim request
+    uint256 public sdReservedForClaim;
+
+    /// @notice minimum block delay between requesting for withdraw and claiming
+    uint256 public undelegationPeriodInBlocks;
+
+    /// @notice minimum block delay between requesting for withdraw and finalization of request
+    uint256 public minBlockDelayToFinalizeRequest;
+
+    /// @notice upper cap on user non redeemed withdraw request count
+    uint256 public maxNonRedeemedDelegatorRequestCount;
+
+    /// @notice address of staderConfig contract
+    IStaderConfig public staderConfig;
+
+    RiskConfig public riskConfig;
+
+    OperatorLiquidation[] public liquidations;
+
+    // Mappings
+    mapping(address => UtilizerStruct) public override utilizerData;
+    mapping(address => uint256) public override delegatorCTokenBalance;
+    mapping(address => uint256) public override delegatorWithdrawRequestedCTokenCount;
+
+    mapping(uint256 => DelegatorWithdrawInfo) public override delegatorWithdrawRequests;
+    mapping(address => uint256[]) public override requestIdsByDelegatorAddress;
+    mapping(address => uint256) private liquidationIndexByOperator;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    //TODO sanjay set variables with right value
+    function initialize(address _admin, address _staderConfig) external initializer {
+        UtilLib.checkNonZeroAddress(_admin);
+        UtilLib.checkNonZeroAddress(_staderConfig);
+        __AccessControl_init_unchained();
+        __Pausable_init();
+        staderConfig = IStaderConfig(_staderConfig);
+        utilizeIndex = DECIMAL;
+        utilizationRatePerBlock = 38051750380; // 10%
+        protocolFeeFactor = 0;
```

Comment:
This should have a denominator, right?

Reply:
Looks like this is a decimal. Is it fair to say that 10% protocol fee would be represented as 10^17?

Reply:
yes, name sound confusing, i will change it